src/index.ts

```diff
@@ -20,18 +20,19 @@ function createThunkMiddleware<
   // Standard Redux middleware definition pattern:
   // See: https://redux.js.org/tutorials/fundamentals/part-4-store#writing-custom-middleware
   const middleware: ThunkMiddleware<State, BasicAction, ExtraThunkArg> =
-    ({ dispatch, getState }) =>
-    next =>
-    action => {
-      // The thunk middleware looks for any functions that were passed to `store.dispatch`.
-      // If this "action" is really a function, call it and return the result.
-      if (typeof action === 'function') {
-        // Inject the store's `dispatch` and `getState` methods, as well as any "extra arg"
-        return action(dispatch, getState, extraArgument)
-      }
+    store => {
```

Comment:
no dereferencing so that when 
```ts
action(store.dispatch, store.getState, extraArgument)
```

is called will use the updated implementation of `store.dispatch` and `store.getState`